src/components/MatrixForm/MatrixForm.tsx

```diff
@@ -1,158 +1,104 @@
 
-	const sum = matrix.map(row => {
-		const sumArr = row.reduce((acc, el) => {
-			return acc + el.amount;
-		}, 0);
+	const matrixRecords = useSelector((state: StoreType) => state.matrixReducer.matrix);
 
-		return sumArr;
-	});
+	const average = useMemo(() => {
+		if (!matrixRecords.length) return null;
+		let sumAr: Array<number> = [];
 
-	const handleAddRow = () => {
-		if (columns && rows) {
-			setRows(rows + 1);
-			setMatrix(prev => [...prev, createMatrix(1, columns).flat()]);
+		for (let i = 0; i < matrixRecords[0].length; i++) {
+			let currentSum = 0;
+			for (let j = 0; j < matrixRecords.length; j++) {
+				currentSum += matrixRecords[j][i].amount;
+			}
+			sumAr = [...sumAr, Math.round(currentSum / matrixRecords.length)];
 		}
-	};
+		return sumAr;
+	}, [matrixRecords]);
+
+	const sumAvg = useMemo(
+		() =>
+			!average
+				? null
+				: average.reduce((acc, el) => {
+						return acc + el;
+				  }, 0),
+		[average]
+	);
 
-	useEffect(() => {
+	const submitSettings = () => {
 		if (columns && rows) {
-			setAverage(avg(matrix, columns, rows));
+			dispatch(setMatrixSettings({ rows, columns, cells }));
 		}
-	}, [matrix]);
+	};
 
 	return (
 		<>
 			<div className={s.matrixForm}>
 				<h2 className={s.matrixForm__title}>Matrix builder</h2>
-				<div className={s.matrixForm__block}>
-					<span className={s.matrixForm__text}>Enter the number of columns</span>
-					<input
-						value={columns || ''}
-						className={s.matrixForm__input}
-						type="number"
-						onChange={e => setColumns(Number(e.target.value))}
-						min="1"
-						max="9999"
-					/>
-				</div>
-				<div className={s.matrixForm__block}>
-					<span className={s.matrixForm__text}>Enter the number of rows</span>
-					<input
-						value={rows || ''}
-						onChange={e => setRows(Number(e.target.value))}
-						className={s.matrixForm__input}
-						type="number"
-						min="1"
-						max="9999"
-					/>
-				</div>
-				<div className={s.matrixForm__block}>
-					<span className={s.matrixForm__text}>Enter the number of cells</span>
-					<input
-						value={cells || ''}
-						onChange={e => setCells(Number(e.target.value))}
-						className={s.matrixForm__input}
-						type="number"
-						min="1"
-						max="9999"
-					/>
-				</div>
+				<SettingField
+					title="Enter the number of columns"
+					value={columns}
+					changeHandler={(columnsValue: number) => setColumns(columnsValue)}
+				/>
+				<SettingField
+					title="Enter the number of rows"
+					value={rows}
+					changeHandler={(rowsValue: number) => setRows(rowsValue)}
+				/>
+				<SettingField
+					title="Enter the number of cells"
+					value={cells}
+					changeHandler={(cellsValue: number) => setCells(cellsValue)}
+				/>
 
-				<button className={s.matrixForm__button} onClick={handleClick} type="button">
+				<button className={s.matrixForm__button} onClick={submitSettings} type="button">
 					Create matrix
 				</button>
 			</div>
 
 			<div>
-				{!!matrix.length && (
```

Comment:
remove matrix from MatrixForm and create component MatrixTable
use ternarochka for matrix length and if no matrix show 'At first, set settings for matrix'